### PR TITLE
RC1 bug fixes and code cleanup

### PR DESCRIPTION
**Pull request by rji**
This PR resolves issues found while testing RC1, specifically around node info being copied / set in individual collectors instead of once when the service starts up. Along with fixing that, this PR also includes a bunch of collector code cleanup to make each collector package more consistent.

There's also a new `collectors.MetricsCollector` interface, but it'll take a bit more work to get the collectors to use it. I'll save that for a future PR.

### CODE REVIEW

File: collectors/node/node.go [hunk 1]
```diff
@@ -26,65 +26,72 @@ import (
...
 	nodeMetrics nodeMetrics
 	timestamp   int64
 }
 
+// New returns a new instance of the node metrics collector and a metrics chan.
+func New(cfg Collector, nodeInfo collectors.NodeInfo) (Collector, chan producers.MetricsMessage) {
+	c := Collector{
+		PollPeriod:  cfg.PollPeriod,
+		MetricsChan: make(chan producers.MetricsMessage),
+		log:         log.WithFields(log.Fields{"collector": "node"}),
```
Comment (malnick) [inline collectors/node/node.go]: :+1:  I really like adding logger to the `New` instance

File: collectors/framework/framework.go [hunk 2]
```diff
@@ -60,100 +64,42 @@ type countingReader struct {
...
-	if err := datapointData.createObjectFromRecord(dps); err != nil {
-		return pmm, err
-	}
-
-	if err := datapointData.extract(&pmm); err != nil {
-		return pmm, err
-	}
-
-	return pmm, nil
+func New(cfg Collector, nodeInfo collectors.NodeInfo) (Collector, chan producers.MetricsMessage) {
```
Comment (malnick) [inline collectors/framework/framework.go]: Really happy to see this as a standalone method.

File: collectors/mesos/agent/agent.go [hunk 3]
```diff
@@ -36,25 +36,20 @@ const (
...
 	RequestProtocol string        `yaml:"request_protocol"`
-
-	MetricsChan chan producers.MetricsMessage
-	NodeInfo    collectors.NodeInfo
-	HTTPClient  *http.Client
+	HTTPClient      *http.Client
 
 	agentState       agentState
 	containerMetrics []agentContainer
+	log              *logrus.Entry
```
Comment (malnick) [inline collectors/mesos/agent/agent.go]: Nit: For people who may be unfamiliar, if might be nice to drop in a comment to documentation `https://godoc.org/github.com/sirupsen/logrus#Entry`

Comment (rji) [inline collectors/mesos/agent/agent.go]: Can do!

**Pull request merged**